python/variable.cpp

```diff
@@ -90,20 +90,25 @@ void bind_init_0D_numpy_types(py::class_<Variable> &c) {
                 b.cast<Eigen::Vector3d>(),
                 v ? std::optional(v->cast<Eigen::Vector3d>()) : std::nullopt,
                 unit);
-          } else if (info.ndim == 1 &&
-                     scipp_dtype(dtype) ==
-                         core::dtype<scipp::core::time_point>) {
-            return do_init_0D<scipp::core::time_point>(
-                b.cast<scipp::core::time_point>(),
-                v ? std::optional(v->cast<scipp::core::time_point>())
-                  : std::nullopt,
-                unit);
           } else if (info.ndim == 2 &&
                      scipp_dtype(dtype) == core::dtype<Eigen::Matrix3d>) {
             return do_init_0D<Eigen::Matrix3d>(
                 b.cast<Eigen::Matrix3d>(),
                 v ? std::optional(v->cast<Eigen::Matrix3d>()) : std::nullopt,
                 unit);
+          } else if ((info.ndim == 1) &&
+                     py::isinstance(b.get_type(), np_datetime64_type)) {
+            // TODO allow construction from int
+            //                     scipp_dtype(dtype) ==
+            //                     core::dtype<core::time_point>) {
+            const auto [actual_unit, value_factor, variance_factor] =
+                get_time_unit(b, v, dtype, unit);
+            return do_init_0D<core::time_point>(
+                make_time_point(b) * value_factor,
+                v.has_value() ? make_time_point(*v) * variance_factor
+                              : std::optional<core::time_point>{},
```

Comment:
Variables with `dtype=time_point` do not support variances. If they did, I suppose the variance would have a different dtype (`int` rather than `time_point`) since `5 pm +/- 1 pm` does not make sense. This is caught at a later level (will throw), but if you check early you can probably simplyfy a bit of code here and in the `get_time_unit` helper?

Reply:
Good point! Removed it